app/src/main/java/eu/kanade/tachiyomi/util/chapter/ChapterSort.kt

```diff
@@ -14,24 +14,33 @@ class ChapterSort(val manga: Manga, val chapterFilter: ChapterFilter = Injekt.ge
         filterForReader: Boolean = false,
         currentChapter: T? = null
     ): List<T> {
+        val filteredChapterList = manga.filtered_scanlators?.let { filteredScanlatorString ->
+            val filteredScanlators = ChapterUtil.getScanlators(filteredScanlatorString)
+            rawChapters.filterNot { ChapterUtil.getScanlators(it.scanlator).any { group -> filteredScanlators.contains(group) } }
+        } ?: rawChapters
```

Comment:
ok so reevalutating, we need this in 3 places:

1.  ChapterSort.getNextUnreadChapter
2. ChapterFilter.filterChaptersForReader
3. ChapterFilter.filterChapters

So we should not have this in ChapterSort.getChaptersSorted, it should be moved to chapter filter

And since this is in 3 places, at this point we should make it into a new method in ChapterFilter

like so
```kotlin
    fun <T : Chapter> filterChaptersByScanlators(chapters: List<T>, manga: Manga): List<T> {
        return manga.filtered_scanlators?.let { filteredScanlatorString ->
            val filteredScanlators = ChapterUtil.getScanlators(filteredScanlatorString)
            chapters.filter { ChapterUtil.getScanlators(it.scanlator).none { group -> filteredScanlators.contains(group) } }
        } ?: chapters
    }
```

then in those 3 cases above we just call ` val filteredChapterList = (chapterFilter.)filterChaptersByScanlators(rawChapters, manga)`

Reply:
It has been done and seems to work as intended.